{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Using this following bash we find 74811 reads in the file we use to test the two tools:\n",
    "\n",
    "grep \">\" test.fa | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "os.getcwd()\n",
    "\n",
    "df = pd.read_table(\"/students/2022-2023/master/creme_de_la_creme/mapseq-test/mapseq-1.2.6-linux/barcode20-mapseq.mseq\", sep='\\t', skiprows=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species count: 74811\n",
      "Unique families: 117\n",
      "Unique species: 466\n"
     ]
    }
   ],
   "source": [
    "print(\"Species count: {}\".format(df['Species'].count()))\n",
    "print(\"Unique families: {}\".format(df['Family'].nunique()))\n",
    "print(\"Unique species: {}\".format(df['Species'].nunique()))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Entrez mapping NCBI refseq identifiers \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Command line option: \n",
    "\n",
    "See ~/Master/creme-de-la-creme/minimap2-blastn/blast.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_blastn = pd.read_table(\"/students/2022-2023/master/creme_de_la_creme/output-blastn/barcode10.txt\", \n",
    "                          sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads as input: 74811\n",
      "\n",
      "Amount of reads mapped to species:  77559\n",
      "Number of species:  77559\n",
      "Amount of unique reads mapped to species:  74811\n",
      "Number of unique species:  1598\n",
      "Number of species not mapped:  0\n"
     ]
    }
   ],
   "source": [
    "print(\"Total number of reads as input: 74811\\n\")\n",
    "\n",
    "print(\"Amount of reads mapped to species: \", df_blastn[0].count())\n",
    "print(\"Number of species: \", df_blastn[12].count())\n",
    "print(\"Amount of unique reads mapped to species: \", df_blastn[0].nunique())\n",
    "print(\"Number of unique species: \", df_blastn[12].nunique())\n",
    "print(\"Number of species not mapped: \", df_blastn[12].isna().sum())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1563"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "family = df_blastn[12].str.split().str.get(0)\n",
    "species = df_blastn[12].str.split().str.get(1)\n",
    "species_blastn = family + ' ' + species\n",
    "species_blastn.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_blastn.columns =['qseqid', 'sseqid', 'pident', 'length', 'mismatch', 'gapopen', 'qstart', 'qend', 'start', 'send', 'evalue', 'bitscore', 'species']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_blastn.columns = pd.MultiIndex.from_tuples(zip(df_blastn.columns,\n",
    "                                                   ['query or source (e.g., gene) sequence id', \n",
    "                                                   'subject  or target (e.g., reference genome) sequence id',\n",
    "                                                   'percentage of identical matches',\n",
    "                                                   'alignment length (sequence overlap)',\n",
    "                                                   'number of mismatches',\n",
    "                                                   'number of gap openings',\n",
    "                                                   'start of alignment in query',\n",
    "                                                   'end of alignment in query',\n",
    "                                                   'start of alignment in subject',\n",
    "                                                   'end of alignment in subject',\n",
    "                                                   'expect value',\n",
    "                                                   'bit score',\n",
    "                                                   'species taxonomic classification']))\n",
    "df_blastn.to_csv('minimap2-blastn/blastn-output-header.csv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>start</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>query or source (e.g., gene) sequence id</th>\n",
       "      <th>subject  or target (e.g., reference genome) sequence id</th>\n",
       "      <th>percentage of identical matches</th>\n",
       "      <th>alignment length (sequence overlap)</th>\n",
       "      <th>number of mismatches</th>\n",
       "      <th>number of gap openings</th>\n",
       "      <th>start of alignment in query</th>\n",
       "      <th>end of alignment in query</th>\n",
       "      <th>start of alignment in subject</th>\n",
       "      <th>end of alignment in subject</th>\n",
       "      <th>expect value</th>\n",
       "      <th>bit score</th>\n",
       "      <th>species taxonomic classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>dd5c53ee-5b3c-46f1-b9a6-99f6c1f25a57</td>\n",
       "      <td>NR_144743.1</td>\n",
       "      <td>87.787</td>\n",
       "      <td>1261</td>\n",
       "      <td>113</td>\n",
       "      <td>34</td>\n",
       "      <td>1</td>\n",
       "      <td>1241</td>\n",
       "      <td>1477</td>\n",
       "      <td>238</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1437.0</td>\n",
       "      <td>Christensenella timonensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>e10378da-0d67-4fd0-9451-cbb01b683de4</td>\n",
       "      <td>NR_037068.1</td>\n",
       "      <td>88.145</td>\n",
       "      <td>1240</td>\n",
       "      <td>98</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>1209</td>\n",
       "      <td>14</td>\n",
       "      <td>1235</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1435.0</td>\n",
       "      <td>Lacrimispora xylanolytica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>06eeff5f-241c-411f-bada-caa19a7a0871</td>\n",
       "      <td>NR_118554.1</td>\n",
       "      <td>86.360</td>\n",
       "      <td>1393</td>\n",
       "      <td>102</td>\n",
       "      <td>60</td>\n",
       "      <td>1</td>\n",
       "      <td>1315</td>\n",
       "      <td>1</td>\n",
       "      <td>1383</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1439.0</td>\n",
       "      <td>Intestinimonas butyriciproducens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8ae3b40d-cc8c-41de-bf9b-a9dd2e8a6f35</td>\n",
       "      <td>NR_026348.1</td>\n",
       "      <td>80.392</td>\n",
       "      <td>1377</td>\n",
       "      <td>202</td>\n",
       "      <td>53</td>\n",
       "      <td>81</td>\n",
       "      <td>1409</td>\n",
       "      <td>1357</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>989.0</td>\n",
       "      <td>Desulforamulus aeronauticus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>07dd2cc4-9dc4-4874-9b66-5a18090023a7</td>\n",
       "      <td>NR_074793.2</td>\n",
       "      <td>90.636</td>\n",
       "      <td>1463</td>\n",
       "      <td>103</td>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>1439</td>\n",
       "      <td>27</td>\n",
       "      <td>1479</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1912.0</td>\n",
       "      <td>Oscillibacter valericigenes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77554</th>\n",
       "      <td>0fddc082-09d9-4651-be1b-fb001027f8fa</td>\n",
       "      <td>NR_026230.1</td>\n",
       "      <td>80.161</td>\n",
       "      <td>1492</td>\n",
       "      <td>203</td>\n",
       "      <td>72</td>\n",
       "      <td>3</td>\n",
       "      <td>1443</td>\n",
       "      <td>1</td>\n",
       "      <td>1450</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1051.0</td>\n",
       "      <td>Porphyromonas catoniae ATCC 51270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77555</th>\n",
       "      <td>7a7bd229-87c4-44a8-94d5-4a525a29654b</td>\n",
       "      <td>NR_118554.1</td>\n",
       "      <td>88.919</td>\n",
       "      <td>1471</td>\n",
       "      <td>118</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>1442</td>\n",
       "      <td>1</td>\n",
       "      <td>1455</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1772.0</td>\n",
       "      <td>Intestinimonas butyriciproducens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77556</th>\n",
       "      <td>a919d02c-509e-4ac6-b1e5-286788c38476</td>\n",
       "      <td>NR_144742.1</td>\n",
       "      <td>84.008</td>\n",
       "      <td>1482</td>\n",
       "      <td>168</td>\n",
       "      <td>57</td>\n",
       "      <td>1</td>\n",
       "      <td>1436</td>\n",
       "      <td>20</td>\n",
       "      <td>1478</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1360.0</td>\n",
       "      <td>Christensenella massiliensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77557</th>\n",
       "      <td>ea7564ae-38a5-429d-8d8f-d1d446bdff47</td>\n",
       "      <td>NR_025931.1</td>\n",
       "      <td>88.714</td>\n",
       "      <td>1338</td>\n",
       "      <td>91</td>\n",
       "      <td>33</td>\n",
       "      <td>83</td>\n",
       "      <td>1374</td>\n",
       "      <td>62</td>\n",
       "      <td>1385</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1589.0</td>\n",
       "      <td>Ruminococcus flavefaciens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77558</th>\n",
       "      <td>186cdc3c-fe50-4d0e-bae5-b3139e8f083a</td>\n",
       "      <td>NR_144613.1</td>\n",
       "      <td>84.138</td>\n",
       "      <td>870</td>\n",
       "      <td>96</td>\n",
       "      <td>34</td>\n",
       "      <td>231</td>\n",
       "      <td>1071</td>\n",
       "      <td>224</td>\n",
       "      <td>1080</td>\n",
       "      <td>0.0</td>\n",
       "      <td>804.0</td>\n",
       "      <td>Irregularibacter muris</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>77559 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        qseqid  \\\n",
       "      query or source (e.g., gene) sequence id   \n",
       "0         dd5c53ee-5b3c-46f1-b9a6-99f6c1f25a57   \n",
       "1         e10378da-0d67-4fd0-9451-cbb01b683de4   \n",
       "2         06eeff5f-241c-411f-bada-caa19a7a0871   \n",
       "3         8ae3b40d-cc8c-41de-bf9b-a9dd2e8a6f35   \n",
       "4         07dd2cc4-9dc4-4874-9b66-5a18090023a7   \n",
       "...                                        ...   \n",
       "77554     0fddc082-09d9-4651-be1b-fb001027f8fa   \n",
       "77555     7a7bd229-87c4-44a8-94d5-4a525a29654b   \n",
       "77556     a919d02c-509e-4ac6-b1e5-286788c38476   \n",
       "77557     ea7564ae-38a5-429d-8d8f-d1d446bdff47   \n",
       "77558     186cdc3c-fe50-4d0e-bae5-b3139e8f083a   \n",
       "\n",
       "                                                       sseqid  \\\n",
       "      subject  or target (e.g., reference genome) sequence id   \n",
       "0                                            NR_144743.1        \n",
       "1                                            NR_037068.1        \n",
       "2                                            NR_118554.1        \n",
       "3                                            NR_026348.1        \n",
       "4                                            NR_074793.2        \n",
       "...                                                  ...        \n",
       "77554                                        NR_026230.1        \n",
       "77555                                        NR_118554.1        \n",
       "77556                                        NR_144742.1        \n",
       "77557                                        NR_025931.1        \n",
       "77558                                        NR_144613.1        \n",
       "\n",
       "                               pident                              length  \\\n",
       "      percentage of identical matches alignment length (sequence overlap)   \n",
       "0                              87.787                                1261   \n",
       "1                              88.145                                1240   \n",
       "2                              86.360                                1393   \n",
       "3                              80.392                                1377   \n",
       "4                              90.636                                1463   \n",
       "...                               ...                                 ...   \n",
       "77554                          80.161                                1492   \n",
       "77555                          88.919                                1471   \n",
       "77556                          84.008                                1482   \n",
       "77557                          88.714                                1338   \n",
       "77558                          84.138                                 870   \n",
       "\n",
       "                  mismatch                gapopen                      qstart  \\\n",
       "      number of mismatches number of gap openings start of alignment in query   \n",
       "0                      113                     34                           1   \n",
       "1                       98                     36                           1   \n",
       "2                      102                     60                           1   \n",
       "3                      202                     53                          81   \n",
       "4                      103                     28                           1   \n",
       "...                    ...                    ...                         ...   \n",
       "77554                  203                     72                           3   \n",
       "77555                  118                     36                           1   \n",
       "77556                  168                     57                           1   \n",
       "77557                   91                     33                          83   \n",
       "77558                   96                     34                         231   \n",
       "\n",
       "                           qend                         start  \\\n",
       "      end of alignment in query start of alignment in subject   \n",
       "0                          1241                          1477   \n",
       "1                          1209                            14   \n",
       "2                          1315                             1   \n",
       "3                          1409                          1357   \n",
       "4                          1439                            27   \n",
       "...                         ...                           ...   \n",
       "77554                      1443                             1   \n",
       "77555                      1442                             1   \n",
       "77556                      1436                            20   \n",
       "77557                      1374                            62   \n",
       "77558                      1071                           224   \n",
       "\n",
       "                             send       evalue  bitscore  \\\n",
       "      end of alignment in subject expect value bit score   \n",
       "0                             238          0.0    1437.0   \n",
       "1                            1235          0.0    1435.0   \n",
       "2                            1383          0.0    1439.0   \n",
       "3                               1          0.0     989.0   \n",
       "4                            1479          0.0    1912.0   \n",
       "...                           ...          ...       ...   \n",
       "77554                        1450          0.0    1051.0   \n",
       "77555                        1455          0.0    1772.0   \n",
       "77556                        1478          0.0    1360.0   \n",
       "77557                        1385          0.0    1589.0   \n",
       "77558                        1080          0.0     804.0   \n",
       "\n",
       "                                 species  \n",
       "        species taxonomic classification  \n",
       "0             Christensenella timonensis  \n",
       "1              Lacrimispora xylanolytica  \n",
       "2       Intestinimonas butyriciproducens  \n",
       "3            Desulforamulus aeronauticus  \n",
       "4            Oscillibacter valericigenes  \n",
       "...                                  ...  \n",
       "77554  Porphyromonas catoniae ATCC 51270  \n",
       "77555   Intestinimonas butyriciproducens  \n",
       "77556       Christensenella massiliensis  \n",
       "77557          Ruminococcus flavefaciens  \n",
       "77558             Irregularibacter muris  \n",
       "\n",
       "[77559 rows x 13 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_blastn"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Kraken2 vs. Bracken"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "kraken2 = pd.read_csv(\"/students/2022-2023/master/creme_de_la_creme/kraken2/barcode05-greengenes/barcode05-greengenes.txt\", sep=\"\\t\", header=None)\n",
    "bracken = pd.read_csv(\"/students/2022-2023/master/creme_de_la_creme/kraken2/barcode05-greengenes/barcode05-greengenes_bracken_species.txt\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Species kraken2:  391982\n",
      "Species Bracken:  414977\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>81.57</td>\n",
       "      <td>339620</td>\n",
       "      <td>339620</td>\n",
       "      <td>S</td>\n",
       "      <td>2556</td>\n",
       "      <td>Propionibacterium acnes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0.01</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>S</td>\n",
       "      <td>2557</td>\n",
       "      <td>Propionibacterium granulosum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>0.00</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>S</td>\n",
       "      <td>2499</td>\n",
       "      <td>Corynebacterium durum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>0.00</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>S</td>\n",
       "      <td>2500</td>\n",
       "      <td>Corynebacterium kroppenstedtii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>0.00</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>S</td>\n",
       "      <td>2506</td>\n",
       "      <td>Corynebacterium stationis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>S</td>\n",
       "      <td>2622</td>\n",
       "      <td>Flavobacterium succinicans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>551</th>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>2630</td>\n",
       "      <td>Sphingobacterium faecium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>2631</td>\n",
       "      <td>Sphingobacterium mizutaii</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>577</th>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>3092</td>\n",
       "      <td>Deinococcus aquatilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>2484</td>\n",
       "      <td>Edaphobacter modestum</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>196 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         0       1       2  3     4  \\\n",
       "7    81.57  339620  339620  S  2556   \n",
       "8     0.01      37      37  S  2557   \n",
       "11    0.00      10      10  S  2499   \n",
       "12    0.00       9       9  S  2500   \n",
       "13    0.00       4       4  S  2506   \n",
       "..     ...     ...     ... ..   ...   \n",
       "542   0.00       2       2  S  2622   \n",
       "551   0.00       1       1  S  2630   \n",
       "552   0.00       1       1  S  2631   \n",
       "577   0.00       1       1  S  3092   \n",
       "588   0.00       1       1  S  2484   \n",
       "\n",
       "                                                5  \n",
       "7                         Propionibacterium acnes  \n",
       "8                    Propionibacterium granulosum  \n",
       "11                          Corynebacterium durum  \n",
       "12                 Corynebacterium kroppenstedtii  \n",
       "13                      Corynebacterium stationis  \n",
       "..                                            ...  \n",
       "542                    Flavobacterium succinicans  \n",
       "551                      Sphingobacterium faecium  \n",
       "552                     Sphingobacterium mizutaii  \n",
       "577                         Deinococcus aquatilis  \n",
       "588                         Edaphobacter modestum  \n",
       "\n",
       "[196 rows x 6 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "species_kraken2 = kraken2[kraken2.loc[:,3] == \"S\"]\n",
    "species_bracken = bracken[bracken.loc[:,3] == \"S\"]\n",
    "\n",
    "print(\"Species kraken2: \", species_kraken2[1].sum())\n",
    "print(\"Species Bracken: \", species_bracken[1].sum())\n",
    "\n",
    "species_kraken2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "6b91fa5cffef32cb10787a50fea9666676a7951181e01280b4644cd70c964bf4"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
